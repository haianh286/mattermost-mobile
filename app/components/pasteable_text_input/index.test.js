// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import {PasteableTextInput} from './index';

describe('PasteableTextInput', () => {
    test('should render pasteable text input', () => {
        const onPaste = jest.fn();
        const text = 'My Text';
        const component = shallow(
            <PasteableTextInput
                onPaste={onPaste}
                screenId='Channel'
            >{text}</PasteableTextInput>,
        );
        expect(component).toMatchSnapshot();
    });

    test('should remove onPaste listener when unmount', () => {
        const mockRemove = jest.fn();
        const text = 'My Text';
        const component = shallow(
            <PasteableTextInput screenId='Channel'>{text}</PasteableTextInput>,
        );

        component.instance().subscription.remove = mockRemove;
        component.instance().componentWillUnmount();
        expect(mockRemove).toHaveBeenCalled();
    });
});
