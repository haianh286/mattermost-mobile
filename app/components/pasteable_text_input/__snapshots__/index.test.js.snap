// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasteableTextInput should render pasteable text input 1`] = `
<TextInput
  onPaste={[Function]}
  screenId="Channel"
>
  My Text
</TextInput>
`;
